### ISSUE TITLE
ICE on two simple (invalid) test programs 

### ISSUE THREAD

**Issue by zhendongsu** [golang/go#16366]
Please answer these questions before submitting your issue. Thanks!

**What version of Go are you using (`go version`)?**

go version go1.6.2 linux/amd64

**What operating system and processor architecture are you using (`go env`)?**

% go env
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH=""
GORACE=""
GOROOT="/home/su/software/go"
GOTOOLDIR="/home/su/software/go/pkg/tool/linux_amd64"
GO15VENDOREXPERIMENT="1"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0"
CXX="g++"
CGO_ENABLED="1"

**What did you do?**

$ go build t1.go
\# command-line-arguments
treecopy [0xc8201a3200]
.   PACK l(3) used(true) main.unsafe
./t1.go:5: internal compiler error: treecopy Name
$ 
$ cat t1.go
package main

import "unsafe"

const _ = unsafe.Sizeof(unsafe{})

$ 
$ go build t2.go
\# command-line-arguments
treecopy [0xc82018f200]
.   PACK l(3) main.unsafe
./t2.go:5: syntax error: unexpected package, expecting name or (
./t2.go:5: internal compiler error: treecopy Name
$ 
$ cat t2.go
package main

import "unsafe"

const _ = unsafe.package (uintptr (1))
$ 

**What did you expect to see?**

both tests rejected with proper error messages 

**What did you see instead?**

internal compiler errors

**Comment by bradfitz** [comment]
I can reproduce at Go 1.6, but both appear to be already fixed at tip (Go 1.7):

```
$ go run t1.go
# command-line-arguments
./t1.go:5: use of package unsafe without selector
./t1.go:5: invalid expression unsafe.Sizeof(composite literal)
$ go run t2.go
# command-line-arguments
./t2.go:5: syntax error: unexpected package, expecting name or (
$ 
$ GOROOT=$HOME/go1.6 go run t1.go
# command-line-arguments
treecopy [0xc820179200]
.   PACK l(3) used(true) main.unsafe
./t1.go:5: internal compiler error: treecopy Name
$ 
$ GOROOT=$HOME/go1.6 go run t2.go
# command-line-arguments
treecopy [0xc820193200]
.   PACK l(3) main.unsafe
./t2.go:5: syntax error: unexpected package, expecting name or (
./t2.go:5: internal compiler error: treecopy Name
```

**Comment by zhendongsu** [comment]
Thanks; that's super fast!
